core/src/main/java/hudson/slaves/CloudProvisioningListener.java

```diff
@@ -61,20 +64,41 @@ public void onComplete(NodeProvisioner.PlannedNode plannedNode, Node node) {
     }
 
     /**
+     * Called when the <code>node</code>is fully connected in the Jenkins.
+     *
+     * @param plannedNode the plannedNode which resulted in the <code>node</code> being provisioned
+     * @param node the node which has been provisioned by the cloud
+     */
+    public void onCommit(NodeProvisioner.PlannedNode plannedNode, Node node) {
+
```

Comment:
Nice to have: Add a comment like "Noop by default". Otherwise tools like Sonar may grumble